### PR TITLE
NET-7153 Proto messages for API Gateway

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description
- Create proto messages for API Gateway

### Testing & Reproduction steps
- This is just the initial structs, will be tested as part of follow up PRs
- Will likely change as part of follow up PRs as well

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: proto-public/pbmesh/v2beta1/api_gateway.proto [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+package hashicorp.consul.mesh.v2beta1;
+
+import "pbresource/annotations.proto";
+import "pbresource/resource.proto";
+
+message APIGateway {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_NAMESPACE};
+
+  // GatewayClassName is the name of the GatewayClass used by the MeshGateway
```
Comment (nathancoleman) [inline proto-public/pbmesh/v2beta1/api_gateway.proto:14]:
```suggestion
  // GatewayClassName is the name of the GatewayClass used by the APIGateway
```

File: proto-public/pbmesh/v2beta1/api_gateway.proto [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  // should support.
+  optional string max_version = 2;
+
+  // MinVersion is the minimum TLS version that the listener
+  // should support.
+  optional string min_version = 3;
+
+  // Define a subset of cipher suites to restrict
+  // Only applicable to connections negotiated via TLS 1.2 or earlier
+  repeated string cipher_suites = 4;
```
Comment (nathancoleman) [inline proto-public/pbmesh/v2beta1/api_gateway.proto:61]: How well does [this pre-existing type](https://github.com/hashicorp/consul/blob/c2a0d4f9ca27aa69ec5555b2f92edff15e0d907c/proto-public/pbmesh/v2beta1/pbproxystate/transport_socket.proto#L110-L141) align with our needs for API gateway cipher suites and min/max TLS versions?

Comment (sarahalsmiller) [reply proto-public/pbmesh/v2beta1/api_gateway.proto:61]: Most likely we can just use them, I'll pull them in.

Comment (sarahalsmiller) [reply proto-public/pbmesh/v2beta1/api_gateway.proto:61]: The one thing they don't seem to have is certificates

File: proto-public/pbmesh/v2beta1/api_gateway.proto [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  string name = 1;
+
+  // +kubebuilder:validation:Maximum=65535
+  // +kubebuilder:validation:Minimum=0
+  uint32 port = 2;
+
+  // Protocol is the protocol that a listener should use, it must
+  // either be "http" or "tcp"
+  // +kubebuilder:validation≈ftg6:Enum=tcp,http
+  string protocol = 3;
```
Comment (jm96441n) [inline proto-public/pbmesh/v2beta1/api_gateway.proto:34]: should we use an actual proto enum here? https://protobuf.dev/programming-guides/proto3/#enum

Comment (sarahalsmiller) [reply proto-public/pbmesh/v2beta1/api_gateway.proto:34]: I believe we need protocol to be a string value, and as far as I can tell from reading the documentation, changing it to a proto enumeration would change the underlying type to an int value, but open to being corrected about this.

Comment (t-eckert) [reply proto-public/pbmesh/v2beta1/api_gateway.proto:34]: That jives with my understanding of how proto enums work.

File: proto-public/pbmesh/v2beta1/api_gateway.proto [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// APIGatewayTLSConfiguration specifies the configuration of a listener’s
+// TLS settings.
+message APIGatewayTLSConfiguration {
+  // Certificates is a set of references to certificates
+  // that a gateway listener uses for TLS termination.
+  repeated hashicorp.consul.resource.Reference certificates = 1;
+
+  // TLSParameters contains
```
Comment (t-eckert) [inline proto-public/pbmesh/v2beta1/api_gateway.proto:52]: I think this comment was left unfinished.

**Review by t-eckert — APPROVED** [review]
Looks solid to me! One comment on an unfinished comment is all I had.

File: proto-public/pbmesh/v2beta1/api_gateway.proto (hunk 4 above)
Comment (sarahalsmiller) [inline proto-public/pbmesh/v2beta1/api_gateway.proto:52]:
```suggestion
  // TLSParameters contains optional configuration for running TLS termination. 
```

**Review by jm96441n — APPROVED** [review]
great work!

**Pull request merged**